packages/batch-submitter/src/batch-submitter/tx-batch-submitter.ts

```diff
@@ -432,6 +439,57 @@ export class TransactionBatchSubmitter extends BatchSubmitter {
       return fixedBatch
     }
 
+    const fixSkippedDeposits = async (b: Batch): Promise<Batch> => {
+      this.logger.debug('Fixing skipped deposits...')
+      let nextQueueIndex = await this.chainContract.getNextQueueIndex()
+      const fixedBatch: Batch = []
+      for (const ele of b) {
+        // Look for skipped deposits
+        while (true) {
+          const pendingQueueElements = await this.chainContract.getNumPendingQueueElements()
+          const nextRemoteQueueElements = await this.chainContract.getNextQueueIndex()
+          const totalQueueElements =
+            pendingQueueElements + nextRemoteQueueElements
+          // No more queue elements so we clearly haven't skipped anything
+          if (nextQueueIndex >= totalQueueElements) {
+            break
+          }
+          const [
+            queueEleHash,
+            timestamp,
+            blockNumber,
+          ] = await this.chainContract.getQueueElement(nextQueueIndex)
+
+          if (timestamp < ele.timestamp && blockNumber < ele.blockNumber) {
```

Comment:
It would be a bug if only one of these was true correct? (exclusive or)

Reply:
Yeah that's true.... Hmm should I just make this a normal OR?

Reply:
Eh, I made it an OR just to be on the safe side